### PR TITLE
Amit/header - Wallet address tokens by list

### PR DESCRIPTION
**Pull request by AmitAyalon**

### CODE REVIEW

File: functions/src/socialAPIs/twitter/getUserTweets.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ const getUserTweets = async (userId: string, order: string) => {
             }
         }
 
-  if (resp && resp.meta && resp.meta.result_count && resp.meta.result_count > 0) {
+  if (resp && resp.data.meta && resp.data.meta.result_count && resp.data.meta.result_count > 0) {
```
Comment (YuvalKrispin) [inline functions/src/socialAPIs/twitter/getUserTweets.ts:33]: add ?
`if (resp && resp.data?.meta && resp.data.meta?.result_count && resp.data.meta.result_count > 0) {`

File: functions/src/usersAPIs/endpoints.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export const getCoinImageEndpoint = async (coinName: string) => {
+  try {
+    const response = await axios.get(`https://api.coingecko.com/api/v3/coins/${coinName}?localization=false&tickers=false&market_data=false&community_data=true&developer_data=false&sparkline=false`, {
+      headers: {
+        "Accept": "application/json",
+      },
+    });
+    return response.data
+  } catch (error) {
+    // console.log(error);
```
Comment (YuvalKrispin) [inline functions/src/usersAPIs/endpoints.ts:22]: remove

File: functions/src/usersAPIs/user-coins.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { getWalletCoinsEndpoint, getCoinImageEndpoint } from "./endpoints";
+interface IWalletCoins {
+    eth: {
+        coinName: string;
+        coinSymbol: string;
+        coinBalance: number;
+        coinPrice: number;
+        coinImage?: string;
+    };
+    tokens: any[];
```
Comment (YuvalKrispin) [inline functions/src/usersAPIs/user-coins.ts:10]: add type

File: functions/src/usersAPIs/user-coins.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        eth: {
+            coinName: "Ethereum",
+            coinSymbol: "ETH",
+            coinBalance: 0,
+            coinPrice: 0,
+            coinImage: "https://api.coingecko.com/api/v3/coins/ethereum",
+        },
+        tokens: [{}],
+    };
+    let response = null;
```
Comment (YuvalKrispin) [inline functions/src/usersAPIs/user-coins.ts:24]: add type

File: functions/src/usersAPIs/user-coins.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    if (response) {
+        if (response.data.ETH.balance) {
+            walletCoins.eth.coinBalance = response.data.ETH.balance;
+            if (response.data.ETH.balance && response.data.ETH.price) {
+                walletCoins.eth.coinPrice = response.data.ETH.balance * response.data.ETH.price.rate;
+            }
+            }
+        const tokens = response.data.tokens;  
+        const tokensArray = [];
```
Comment (YuvalKrispin) [inline functions/src/usersAPIs/user-coins.ts:39]: add type

File: functions/src/usersAPIs/user-coins.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        if (tokens.length > 0) {
+            for (let i = 0; i < tokens.length; i++) {
```
Comment (YuvalKrispin) [inline functions/src/usersAPIs/user-coins.ts:41]: why not foreach?

File: src/components/CoinSelection/index.tsx [hunk 7]
```diff
@@ -18,32 +15,65 @@ const CoinSelection = observer(() => {
...
     </div>
   );
-
-})
+});
 export default CoinSelection;
 
-const CoinListItem = observer(({ coin }: { coin: ICoinData; }) => { // gal it seems this is not working with mobx
+const CoinListItem = observer(({ coin }: { coin: ICoinData }) => {
+  // gal it seems this is not working with mobx
```
Comment (YuvalKrispin) [inline src/components/CoinSelection/index.tsx:42]: remove my comment please

File: src/components/wallet-coins/WalletCoins.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+          <SearchOutlined
+            onClick={() => {
+              userStore.getWalletTokens(searchInput);
+            }}
+          />
+        }
+        size="large"
+        onClick={() => setDisplayList(true)}
+      />
+      <div className="wallet-coin-list" style={{ color: 'white' }}>
```
Comment (YuvalKrispin) [inline src/components/wallet-coins/WalletCoins.tsx:47]: move style to class

File: src/store/user-store/index.ts [hunk 9]
```diff
@@ -11,6 +11,8 @@ class UserStore {
     password: ''
   };
 
+  walletTokens: any = {};
```
Comment (YuvalKrispin) [inline src/store/user-store/index.ts:14]: type

File: src/store/user-store/index.ts [hunk 10]
```diff
@@ -46,6 +48,19 @@ class UserStore {
     });
   }
 
+  getWalletTokens(walletAddress: string) { 
+    functionsCall('getWalletCoins')(walletAddress).then((result: any) => {
+      console.log(result);
```
Comment (YuvalKrispin) [inline src/store/user-store/index.ts:53]: remove console.log

File: src/store/user-store/index.ts [hunk 11]
```diff
@@ -46,6 +48,19 @@ class UserStore {
     });
   }
 
+  getWalletTokens(walletAddress: string) { 
+    functionsCall('getWalletCoins')(walletAddress).then((result: any) => {
```
Comment (YuvalKrispin) [inline src/store/user-store/index.ts:52]: add type

File: functions/src/usersAPIs/user-coins.ts (hunk 6 above)
Comment (AmitAyalon) [reply functions/src/usersAPIs/user-coins.ts:52]: doesn't get the data with foreach, map somehow only traditional for loop does the job

**Pull request merged**